### PR TITLE
Remove "wait" from registry deployment

### LINKED ISSUE #85: Replace docker login with local generated file
We can skip the task of waiting for the registry to come up in the registry role, and save several minutes, if we don't actually attempt a docker login against the registry, but instead construct the file using templating.

### PR DESCRIPTION
**Pull request by aperullo**
Closes #85

The deployment of the registry had a step that had to wait for the registry to come up before it could `docker login`. This change removes the wait and the docker login command, replacing it with what the resulting file would be on success. This cuts several minutes off the deploy time and is one less point of failure.

My one concern is that the config file is overwritten regardless, which is why I create a backup first. The file should be empty on a new node, but I couldn't find a good way to merge it with one if it already exists without pulling in several more dependencies.

### CODE REVIEW

File: kubernetes/ansible/playbooks/roles/registry_role/templates/config.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+	"auths": {
+		"https://registry.rke.gsp.test": {
```
Comment (njbarber) [inline kubernetes/ansible/playbooks/roles/registry_role/templates/config.json]: Should this be relative to the current workspace i.e. `https://{{ ingress_host }}`?

File: kubernetes/ansible/playbooks/roles/registry_role/tasks/install.yml [hunk 2]
```diff
@@ -144,13 +144,10 @@
...
-  --username={{ harbor_user }} \
-  --password={{ hostvars[groups['kube-setup-delegate'][0]]['harbor_pass'] }} \
-  'https://{{ ingress_host }}'"
-  no_log: true    # Prevent the credentials from showing in the logs
+- name: Registry | Template docker config.json
+  template:
+    src: "config.json"
+    dest: "~/.docker/config.json"
+    backup: yes
+  no_log: true    # Prevent the credentials from showing in the logs
```
Comment (njbarber) [inline kubernetes/ansible/playbooks/roles/registry_role/tasks/install.yml]: Does docker need to be restarted after this?

File: kubernetes/ansible/playbooks/roles/registry_role/templates/config.json (hunk 1 above)
Comment (aperullo) [reply kubernetes/ansible/playbooks/roles/registry_role/templates/config.json]: Oh very good catch yes it should

File: kubernetes/ansible/playbooks/roles/registry_role/tasks/install.yml (hunk 2 above)
Comment (aperullo) [reply kubernetes/ansible/playbooks/roles/registry_role/tasks/install.yml]: No, it reads the config when the next operation is called.

**Pull request merged**